WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -6,178 +11,539 @@ Set-PSDebug -Strict
-    $wslstatus = wsl -l -v
-    if($LASTEXITCODE -eq 1)
+    Intall-WSLBlobNFS-Internal
+    if($LASTEXITCODE -ne 0)
     {
-        Write-Host "WSL is not installed. Installing WSL..."
-
-        wsl --install -d Ubuntu-22.04
-        Write-Host "Restart the VM to complete WSL installation, and then Run the script again with Initialize-WSLBlobNFS to continue the WSL setup."
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
         return
     }
 
-    # Run the script with setupwslenv argument
-    # To-do: Automate to start this on startup.
+    $initialized = $false
+    if($Force)
+    {
+        Write-Warning "Force initializing WSL environment for WSLBlobNFS usage."
+    }
+    else
+    {
+        Write-Debug "Initializing WSL environment for your WSLBlobNFS usage."
+    }
 
-    Write-Host "Installing Ubuntu-22.04..."
-    wsl --install -d Ubuntu-22.04
+    # # Add the modulePathForLinux, which hold the path of module, to WSLENV to help in copying linux scripts to wsl distro.
+    # [Environment]::SetEnvironmentVariable("modulePathForLinux", $PSScriptRoot)
+    # $p = [Environment]::GetEnvironmentVariable("WSLENV")
+    # $p += ":modulePathForLinux/p"
+    # [Environment]::SetEnvironmentVariable("WSLENV",$p)
 
-    # ask user if the username was setup
-    $usernameSetup = Read-Host "Was the wsl distro username setup? (y/n)"
-    if($usernameSetup -ne "y")
+    $moduleVersionOnWin = Get-ModuleVersion
+    Write-Debug "Module version on Windows: $moduleVersionOnWin"
+    $moduleVersionOnWSL = Execute-WSL "cat $linuxScriptsPath/$versionFileName 2>/dev/null"
+    Write-Debug "Module version on WSL: $moduleVersionOnWSL"
+    if(($LASTEXITCODE -eq 0) -and ($moduleVersionOnWSL.Trim() -eq $moduleVersionOnWin) -and !$Force)
     {
-        Write-Host "Restart the VM to complete wsl installation, and then run the script again with setupwslenv action to continue the wsl setup."
-        return
+        Write-Debug "WSL Files are upto date. Skipping copying files."
+    }
+    else
+    {
+        $initialized = $true
+        if($Force)
+        {
+            Write-Debug "Force parameter is provided. Copying files again."
+        }
+        elseif ($LASTEXITCODE -ne 0)
+        {
+            Write-Debug "Module is not installed. Copying files again."
+        }
+        else
+        {
+            Write-Debug "Module version is changed. Copying files again."
+        }
+
+        Copy-WSLFiles
+        if ($LASTEXITCODE -ne 0)
+        {
+            Write-Error "Copying wsl files failed."
+            return
+        }
+
+        Execute-WSL "echo $moduleVersionOnWin > $linuxScriptsPath/$versionFileName"
     }
 
-    Write-Host "Setting up the WSL environment for your WSLBlobNFS usage."
+    # Install systemd, restart wsl and install NFS & Samba
+    Execute-WSL "systemctl list-unit-files --type=service | grep systemd > /dev/null 2>&1"
```

Comment:
to ensure we don't match some other package/service with systemd in its name, let's grep for '^systemd-'